src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,119 @@
-function rottenTomatoes(grid) {}
+//declaration and initialization of global variables
+const freshTomatoe = 1;
+const rottenTomatoe = 2;
+let timeToRotInMinute;
+let tomatoes = [];
 
+function rottenTomatoes(grid) 
+{
+    //check to ensure input is a valid parameter
+    if (!Array.isArray(grid) || grid.length === 0)
+    {
+        throw `Please enter a valid input`;
```

Comment:
We covered this before.

Prefer to `throw Error(...)` rather than strings. Helps preserve the stack trace for debugging.

Reply:
To this effect, sir, would something like this serve the purpose?
try
	{
		if (!Array.isArray(tomatoes) || tomatoes.length === 0)
		{
			throw new Error(`Please enter a valid input`);
		}
	}
	catch (errorMessage)
	{
		console.log(errorMessage);
	}

I had wanted to throw the error and not catch it to be sure I grasp the last review you made on error throwing. Just hope I have not done otherwise...

Reply:
No need to catch the error. Simply replacing that line with

```js
throw Error('Please enter a valid input');
```

Is what I mean. Look at the feedback messages on the board for previous weeks.

Reply:
ok